modeling-cmds/src/shared.rs

```diff
@@ -411,6 +411,16 @@ where
     }
 }
 
+///A quaternion
+pub type Quaternion = Point4d;
```

Comment:
I suggest quarternion should be a newtype which wraps point4d, not an alias of point4d. This way users cannot arbitrarily intermix quarternions with normal point4ds. Also it allows you to define a `default` method for quarternion but not for all other point4ds.

```suggestion
pub struct Quaternion(Point4d);
```